pkg/controller/machine_bootstrap_token.go

```diff
@@ -0,0 +1,226 @@
+	if err != nil {
+		return err
+	}
+
+	token := bootstraptokenutil.TokenFromIDAndSecret(
+		string(bootstrapTokenSecret.Data[bootstraptokenapi.BootstrapTokenIDKey]),
+		string(bootstrapTokenSecret.Data[bootstraptokenapi.BootstrapTokenSecretKey]),
+	)
+	glog.V(4).Infof("Add bootstrap token %q to user-data!", token)
+	userData, err = addBootsrapTokenToUserData(userData, token)
+	if err != nil {
+		return err
+	}
+	driver.SetUserData(userData)
+	return nil
+}
+
+func (c *controller) computeBootstrapToken(machineName string) (secret *v1.Secret, err error) {
+	tokenID, secretName := getTokenIDAndSecretName(machineName)
+
+	secret, err = c.targetCoreClient.CoreV1().Secrets(metav1.NamespaceSystem).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			bootstrapTokenSecretKey, err := generateRandomStringFromCharset(16, "0123456789abcdefghijklmnopqrstuvwxyz")
+			if err != nil {
+				return nil, err
+			}
+			data := map[string][]byte{
+				bootstraptokenapi.BootstrapTokenDescriptionKey:      []byte("A bootstrap token generated by MachineControllManager."),
+				bootstraptokenapi.BootstrapTokenIDKey:               []byte(tokenID),
+				bootstraptokenapi.BootstrapTokenSecretKey:           []byte(bootstrapTokenSecretKey),
+				bootstraptokenapi.BootstrapTokenExpirationKey:       []byte(metav1.Now().Add(48 * time.Hour).Format(time.RFC3339)),
+				bootstraptokenapi.BootstrapTokenUsageAuthentication: []byte("true"),
+				bootstraptokenapi.BootstrapTokenUsageSigningKey:     []byte("true"),
+			}
+
+			secret = &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: metav1.NamespaceSystem,
+				},
+				Type: bootstraptokenapi.SecretTypeBootstrapToken,
+				Data: data,
+			}
+
+			return c.targetCoreClient.CoreV1().Secrets(metav1.NamespaceSystem).Create(secret)
+		}
+		return nil, err
+	}
+
+	return secret, nil
+}
+
+func (c *controller) deleteBootstrapToken(machineName string) error {
+	_, secretName := getTokenIDAndSecretName(machineName)
+	err := c.targetCoreClient.CoreV1().Secrets(metav1.NamespaceSystem).Delete(secretName, &metav1.DeleteOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
+		// Object no longer exists and has been deleted
+		return nil
+	}
+	return err
+}
+
+func addBootsrapTokenToUserData(userdata, token string) (string, error) {
+	config := map[string]interface{}{
+		"PATH":    bootstrapTokenPath,
+		"DIR":     dirKubelet,
+		"CONTENT": base64.StdEncoding.EncodeToString([]byte(token)),
+	}
+
+	userdata = strings.TrimSpace(string(userdata))
+	if strings.HasPrefix(userdata, "#cloud-config") {
+		return execTemplateAndAddItToUserData("CLOUD_CONFIG_WRITE_FILE", config, userdata, addWriteFileToCloudConfig)
+	} else if match, _ := regexp.MatchString("^#\\s*!.*\\/(bash|sh|env)", userdata); match {
+		return execTemplateAndAddItToUserData("SCRIPT_WRITE_FILE", config, userdata, addWriteFileToCloudConfig)
```

Comment:
```suggestion
		return execTemplateAndAddItToUserData("SCRIPT_WRITE_FILE", config, userdata, addWriteFileToScript)
```

Reply:
OMG how stupid I am.
it is removed from the code